### ISSUE TITLE
Style/SpecialGlobalVars : $INPUT_RECORD_SEPARATOR and $RS not identical to $/

### ISSUE THREAD

**Issue by tilo** [rubocop/rubocop#10733]
# Style/SpecialGlobalVars 

* potentially incorrect rule 
* should be marked unsafe

* `$INPUT_RECORD_SEPARATOR` and `$RS` are not identical to `$/`

* this rule introduces bugs in some cases

## Expected behavior

Rubocop should only flag safe replacements
 
## Actual behavior

```
 2.7.5-p203 :001 > RUBY_VERSION
 => "2.7.5"
2.7.5-p203 :002 > $/
 => "\n"
2.7.5-p203 :003 > $RS
 => nil
2.7.5-p203 :004 > $INPUT_RECORD_SEPARATOR
 => nil
 ```

## Rubocop Version

```
1.30.1 (using Parser 3.1.2.0, rubocop-ast 1.18.0, running on ruby 2.7.5 x86_64-darwin21)
```

## OS

Mac OSX 12.4

**Comment by r7kamura** [comment]
@tilo 

[The Style/SpecialGlobalVars document](https://docs.rubocop.org/rubocop/cops_style.html#stylespecialglobalvars) says:

> This cop looks for uses of Perl-style global variables. Correcting to global variables in the 'English' library will add a require statement to the top of the file if enabled by RequireEnglish config.

Maybe you forgot to call `require 'English'` ?

```
irb(main):001:0> RUBY_VERSION
=> "2.7.5"
irb(main):002:0> $/
=> "\n"
irb(main):003:0> $RS
=> nil
irb(main):004:0> $INPUT_RECORD_SEPARATOR
=> nil
irb(main):005:0> require 'English'
=> true
irb(main):006:0> $RS
=> "\n"
irb(main):007:0> $INPUT_RECORD_SEPARATOR
=> "\n"
```

**Comment by jonas054** [comment]
The cop already reminds the user to add `require 'English'`. For example:
```
file.rb:3:3: C: [Correctable] Style/SpecialGlobalVars: Prefer $INPUT_RECORD_SEPARATOR or $RS from the stdlib 'English' module (don't forget to require it) over $/.
p $/
  ^^
```
Its default configuration is:
```yaml
Style/SpecialGlobalVars:
  Description: 'Avoid Perl-style global variables.'
  StyleGuide: '#no-cryptic-perlisms'
  Enabled: true
  VersionAdded: '0.13'
  VersionChanged: '0.36'
  SafeAutoCorrect: false
  RequireEnglish: true
  EnforcedStyle: use_english_names
  SupportedStyles:
    - use_perl_names
    - use_english_names
    - use_builtin_english_names
```
which means that it's marked as a safe cop, but with unsafe autocorrection. Cop documentation explains this:

>  Autocorrection is marked as unsafe because if `RequireEnglish` is not true, replacing perl-style variables with english variables will break.

So with _default_ configuration, autocorrection is actually safe.
